src/MDOP.jl

```diff
@@ -4,16 +4,150 @@ using Distances
+  the `τs`'s the β-statistic gets computed.
+* `w::Int = 1`: Theiler window (neighbors in time with index `w` close to the point,
+  that are excluded from being true neighbors). `w=0` means to exclude only the
+  point itself, and no temporal neighbors.
+* `fnn_thres::Float64= 0.05`: A threshold value defining a sufficiently small fraction
+  of false nearest neighbors, in order to the let algorithm terminate and stop
+  the embedding procedure (`0 > fnn_thres > 1).
+* `r::Float64 = 2.0`: The threshold for the tolerable relative increase of the distance
+  between the nearest neighbors, when increasing the embedding dimension.
+* `βs::Bool = false`: When set `true`, the function also returnes the `β`-statistics
+  of all embedding cycles.
+
+## Description
+The method works iteratively and gradually builds the final embedding vectors
+`Y`. Based on the [`beta_statistic`](@ref) the algorithm picks an optimal delay
+value `τ` for each embedding cycle as the global maximum of `β`. In case of
+multivariate embedding, i.e. when embedding a set of time series (`s::Dataset`),
+the optimal delay value `τ` is chosen as the maximum from all maxima's of all
+considered `β`-statistics for each embedding cycle. The range of
+considered delay values is determined in `τs` and for the nearest neighbor
+search we respect the Theiler window `w`. After each embedding cycle the FNN-
+statistic `FNNs` [^Hegger1999][^Kennel1992] is being checked and as soon as this statistic
+drops below the threshold `fnn_thres`, the algorithm breaks. In order to
+increase the  practability of the method the algorithm also breaks, when the
+FNN-statistic `FNNs` increases . The final embedding vector is stored in `Y`
+(`Dataset`). The chosen delay values for each embedding cycle are stored in the
+`τ_vals` and the according time series number chosen for the according delay
+value in `τ_vals` is stored in `ts_vals`. For univariate embedding (`s::Vector`)
+`ts_vals` is a vector of ones of length `τ_vals`, because there is simply just
+time series to choose from. If `βs=true` then the function also returns the
+`β`-statistic for each embedding cycle as a `List` of `Vector`s.
 
 [^Nichkawde2013]: Nichkawde, Chetan (2013). [Optimal state-space reconstruction using derivatives on projected manifold. Physical Review E 87, 022905](https://doi.org/10.1103/PhysRevE.87.022905).
+[^Hegger1999]: Hegger, Rainer and Kantz, Holger (1999). [Improved false nearest neighbor method to detect determinism in time series data. Physical Review E 60, 4970](https://doi.org/10.1103/PhysRevE.60.4970).
+[^Kennel1992]: Kennel, M. B., Brown, R., Abarbanel, H. D. I. (1992). [Determining embedding dimension for state-space reconstruction using a geometrical construction. Phys. Rev. A 45, 3403] (https://doi.org/10.1103/PhysRevA.45.3403).
 """
-function MDOP(Y...)
+function MDOP(s::Vector{Float64}; τs = 0:50 , w::Int = 1, fnn_thres::Float64 = 0.05,
+    r::Float64 = 2.0, βs::Bool=false)
```

Comment:
Notice that type asserting `Float64` is not a good idea: it makes your code not work with other numeric types. What you want to do instead is assert as follows:
```julia
s::Vector{<: Real} # vector of any real numbers
r::Real # any real number
```
This is **not** type unstable. It just allows more types while still annotating that you would expect a vector of numbers.